go/host/host.go

```diff
@@ -1025,26 +1039,57 @@ func (h *host) checkBlockForSecretResponse(block *types.Block) bool {
 	return false
 }
 
-// Handles an incoming batch.
-func (h *host) handleBatch(encodedBatch *common.EncodedBatch) error {
-	batch := common.ExtBatch{}
-	err := rlp.DecodeBytes(*encodedBatch, &batch)
+// Handles an incoming set of batches.
+func (h *host) handleBatches(encodedBatches *common.EncodedBatches) error {
+	var batches []*common.ExtBatch
+	err := rlp.DecodeBytes(*encodedBatches, &batches)
 	if err != nil {
-		return fmt.Errorf("could not decode batch using RLP. Cause: %w", err)
+		return fmt.Errorf("could not decode batches using RLP. Cause: %w", err)
+	}
+	if len(batches) == 0 {
+		return nil
 	}
 
-	// TODO - #718 - Have the enclave process batch, so that it's up to date.
-
-	// TODO - #718 - Implement a catch-up mechanism for historical batches.
+	// TODO - #718 - Have the enclave process the batch, so that it's up to date.
 
-	err = h.db.AddBatchHeader(&batch)
+	// We request any batches we've missed.
+	batchRequest, err := h.batchManager.CreateBatchRequest(batches)
```

Comment:
I find this whole flow very confusing. We've just received some batches I think, and then we call createBatchRequest with those batches. If you get rid of the sanity checks on incoming batches for now does it all still work? I think you're creating confusion by trying to fend off attacks that might not really exist.

Reply:
I've renamed it for clarity - it's now `IsMissingBatches`, which returns a bool indicating if any historical batches are missing, plus the batch request needed to retrieve them.

All this would be cleaner if `BatchManager` had a reference to the P2P component, but it felt better overall to make all P2P calls go via the host. Thoughts?